extras/drift_postgres/lib/src/pg_database.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:collection/collection.dart';
+import 'package:drift/backends.dart';
+import 'package:postgres/postgres.dart';
+
+///
+class PgDatabase extends DelegatedDatabase {
+  ///
+  PgDatabase(PostgreSQLConnection connection)
+      : super(_PgDelegate(connection, connection),
+            isSequential: true, logStatements: true);
+}
+
+class _PgDelegate extends DatabaseDelegate {
+  final PostgreSQLConnection _db;
+  final PostgreSQLExecutionContext _ec;
+
+  @override
+  SqlDialect get dialect => SqlDialect.postgres;
+
+  _PgDelegate(this._db, this._ec) : closeUnderlyingWhenClosed = false;
+
+  bool _isOpen = false;
+
+  final bool closeUnderlyingWhenClosed;
+
+  @override
+  TransactionDelegate get transactionDelegate => _PgTransactionDelegate(_db);
+
+  @override
+  late DbVersionDelegate versionDelegate;
+
+  @override
+  Future<bool> get isOpen => Future.value(_isOpen);
+
+  @override
+  Future<void> open(QueryExecutorUser user) async {
+    final pgVersionDelegate = _PgVersionDelegate(_db);
+
+    await _db.open();
+    await pgVersionDelegate.init();
+
+    versionDelegate = pgVersionDelegate;
+    _isOpen = true;
+  }
+
+  Future _ensureOpen() async {
+    if (_db.isClosed) {
+      await _db.open();
+    }
+  }
+
+  final _regexIndexed = RegExp(r'\?(\d+)');
+  final _regexNamed = RegExp(r':(\w+)');
+  final _escapeChar = String.fromCharCode(0x0);
+
+  String _convertStatement(String statement) {
```

Comment:
FWIW, we can also have this depend on `sqlparser` directly and use the tokenizer from that package to identify variable tokens.

Even without adding new token types to that package, you can do `Scanner(statement).scanTokens()` and then replace `QuestionMarkVariableToken`s with `@` tokens. Replacing could work by looking at the token's `span`, which reports the region of the string to replace.

Reply:
Thank you for suggestion. I can't know and use the parser yet :) This would impact query performance, I think placeholder should be replaced anyways (in long term). May we can use a config for placeholder mark?
In deed, this question mark problem affects only custom queries. Drift could choose mark depending on dialect in generated queries (one of my older code did this). For SQLite nothing change, custom queries will run as before. However, if somebody want to use Postgres, he have to use or change '?' marks to '@'/'$'. SQLite will understand both.

Reply:
Tokenization only should be reasonably fast (I hope it's comparable to RegExps, otherwise I'll have to take a look at the tokenizer to see where it's going wrong), but indeed there is going to be some overhead.

> However, if somebody want to use Postgres, he have to use or change '?' marks to '@'/'$'. SQLite will understand both.

I think this is fine initially. The SQL parser used to analyze queries is heavily focused on sqlite, so it will take a while to have good support for other dialects there. We have a build option that rewrites queries at compile-time, so we could have another dialect option to rewrite variables too. We don't need to do this in this PR though, let's just rewrite variables at runtime for now (either with the tokenizer or with regexps)

Reply:
I think tokenizer is faster than regex, I compared overhead to without rewriting. But, I think better to add a runtime option to rewrite or not. For ex, I want to use Postgres and I only use '@' because of that. In this case tokenizer runs for nothing unnecessarily. This parameter mark issue not affects builder, only runtime. I suppose to add a 'compatible mode' option to Postgres connection/PgDatabase. When it is true, mark stay '?' everywhere as currently and runs tokenizer. If false, parameter mark depends on dialect in generated queries.

Reply:
I like that idea 👍